.azure-pipelines/code-oob.yml

```diff
@@ -1,70 +1,44 @@
-  type: string
-  default: main
 - name: ServiceName
-  displayName: Service to OOB release
+  displayName: OOB release Service
   type: string
   default: Databricks
 - name: Preview
-  displayName: Is preview version
+  displayName: Is preview version(Including < 1.0.0)
   type: boolean
   default: false
+variables:
+  BumpUpBranch: none
 stages:
 - stage: OOBRelease
   displayName: OOBRelease
   jobs:
-  - job: OOBRelease
+  - job: BumpUpVersion
     timeoutInMinutes: 180
     pool: pool-windows-2019
+    variables:
+      NeedBumpUp: true
     steps:
-    - task: DotNetCoreCLI@2
-      displayName: Build
-      inputs:
-        command: custom
-        custom: msbuild
-        arguments: 'build.proj "/t:Clean;Build" /p:Configuration=Release'
-    - pwsh: |
-          .\tools\RunVersionController.ps1 -ModuleName "Az.${{ parameters.ServiceName }}”
-      displayName: 'Bump up version'
-    - task: AzurePowerShell@5
-      inputs:
-          azureSubscription: '$(AzureSubscription)'
-          ScriptType: 'InlineScript'
-          azurePowerShellVersion: 'LatestVersion'
-          inline: |
-              Set-AzContext -Tenant $(TenantId) -SubscriptionId $(SubscriptionId)
-              $Token = Get-AzKeyVaultSecret -VaultName $(KeyVaultName) -Name $(KeyVaultAccount) -AsPlainText
-              git config user.email "65331932+azure-powershell-bot@users.noreply.github.com"
-              git config user.name "azure-powershell-bot"
-              git checkout -b codeoob/${{ parameters.ServiceName }}
-              git add -A
-              if (${{ parameters.Preview }} -eq $true) {
-                git reset -- ./tools/Tools.Common/SerializedCmdlets/Az.${{ parameters.ServiceName }}.json
-              }
-              git commit -m "Bump up version for ${{ parameters.ServiceName }}"
-              git remote set-url origin https://azure-powershell-bot:$Token@github.com/Azure/azure-powershell.git;
-              git push origin codeoob/${{ parameters.ServiceName }} --force;
-              $Title = "Migrate ${{ parameters.ServiceName }} from oob to main"
-              $HeadBranch = "codeoob/${{ parameters.ServiceName }}"
-              $BaseBranch = "${{ parameters.TargetBranch }}"
-              $Description = "Migrate ${{ parameters.ServiceName }} from oob to ${{parameters.TargetBranch}}"
-              ./tools/Github/CreatePR.ps1 -Title $Title -HeadBranch $HeadBranch -BaseBranch $BaseBranch -BotAccessToken $Token -Description $Description -Draft $true
-      displayName: Create PR to main branch
     - task: PowerShell@2
       displayName: get changelog info
+      name: outputinfo
       inputs:
         targetType: 'inline'
         script: |
           $readme = Get-ChildItem -Path "./src/${{ parameters.ServiceName }}" -name ChangeLog.md -Recurse
           $mdContent = Get-Content -Path "src/${{ parameters.ServiceName }}/$readme" -Raw
           $pattern = '(?s)## Version (\d+\.\d+\.\d)+[\s\S]*?\n(.*?)(?=\n## Version \d+\.\d+\.\d+|$)'

```

Comment:
How about we simplify the regex and leverage `[System.Version]` to parse the version string? Complex regex is hard to maintain and is easy to miss some corner case (like what if there are two spaces between "##" and "Version").

Reply:
Good suggestion. I would like to try. Actually, this is a relatively complicated situation. It is necessary to consider both multiple lines and only one '## version'.